upgrade[patch]: @microsoft/teamsfx-cli to v1.2.1

JensMadsen opened the pull request:
resolves security vulnerability from jsonwebtoken package < 9.0.0

**Change Description**:

package upgrade which also resolves security vulnerability from an older version of the jsonwebtoken package.

I am unsure what to do with the lock files? 

1. **Do these changes impact *command syntax* of any of the packages?** (e.g., add/remove command, add/remove a command parameter, or update required parameters)
No


2. **Do these changes impact *documentation*?** (e.g., a tutorial on https://docs.microsoft.com/en-us/office/dev/add-ins/overview/office-add-ins)
No


If you answered yes to any of these please do the following:
    > Include 'Rick-Kirkham' in the review
    > Make sure the README file is correct

**Validation/testing performed**:

I ran test suite

millerds commented:
/Azurepipelines run

millerds reviewed (changes requested):
Please include an updated package-lock.json file . . . pipelines will not pass w/out it.

millerds commented:
I also tried this out locally and it doesn't seem to fix the jsonwebtoken vulnerability . . . it's still reported when running 'npm audit'.  Looks like the @azure/* modules need to be updated to support a fixed version of jsonwebtoken before other upstream modules can be corrected.

millerds commented:
Lerna is a tool used to manage multiple packages in a single repo.  Each folder under the packages folder gets published as it's own module.

I'm familiar with occasionally needing to regenerate the lock file and then node_modules directory.  In this case I went into office-addin-dev-settings where the change needs to happen, removed the node_modules directory and the package-lock.json file, updated the package.json file with the new version of teamsfx-cli, and ran 'npm install'.  This regenerated the package-lock.json file (yes, a lot of changes) and installed all the modules, and it still reported the vulnerability.  Running 'npm audit fix' does not correct it either and reports 'no fix available' as well.

I know there are differences between how yarn and npm handle things, but I'm not sure why there would be a difference in vulnerability reporting.

JensMadsen commented:
> Lerna is a tool used to manage multiple packages in a single repo. Each folder under the packages folder gets published as it's own module.
> 
> I'm familiar with occasionally needing to regenerate the lock file and then node_modules directory. In this case I went into office-addin-dev-settings where the change needs to happen, removed the node_modules directory and the package-lock.json file, updated the package.json file with the new version of teamsfx-cli, and ran 'npm install'. This regenerated the package-lock.json file (yes, a lot of changes) and installed all the modules, and it still reported the vulnerability. Running 'npm audit fix' does not correct it either and reports 'no fix available' as well.
> 
> I know there are differences between how yarn and npm handle things, but I'm not sure why there would be a difference in vulnerability reporting.

I see what you mean. It seems to be more complicated than I thought. A lot of the problems come down to outdated versions of the @azure/identity package e.g. in the `dev-settings` package the @microsoft/teamsfx-core and the @microsoft/teamsfx-cli packages should be updated. `npm audit fix` yields
``` sh 
yolen@yolen-ThinkPad-P14s-Gen-2a ~/Desktop/Office-Addin-Scripts/packages/office-addin-dev-settings (upgradeTeamsFxCli)$ npm audit fix

up to date, audited 953 packages in 2s

149 packages are looking for funding
  run `npm fund` for details

# npm audit report

jsonwebtoken  <=8.5.1
Severity: high
jsonwebtoken unrestricted key type could lead to legacy keys usage  - https://github.com/advisories/GHSA-8cf7-32gw-wr33
jsonwebtoken has insecure input validation in jwt.verify function - https://github.com/advisories/GHSA-27h2-hvpr-p74q
jsonwebtoken's insecure implementation of key retrieval function could lead to Forgeable Public/Private Tokens from RSA to HMAC - https://github.com/advisories/GHSA-hjrf-2m68-5959
jsonwebtoken vulnerable to signature validation bypass due to insecure default algorithm in jwt.verify() - https://github.com/advisories/GHSA-qwph-4952-7xr6
No fix available
node_modules/jsonwebtoken
  @azure/msal-node  *
  Depends on vulnerable versions of jsonwebtoken
  node_modules/@azure/msal-node
  node_modules/@microsoft/teamsfx-core/node_modules/@azure/msal-node
  node_modules/tedious/node_modules/@azure/msal-node
    @azure/identity  >=1.2.0-alpha.20200903.1
    Depends on vulnerable versions of @azure/msal-node
    node_modules/@azure/identity
    node_modules/tedious/node_modules/@azure/identity
      @microsoft/teamsfx-cli  *
      Depends on vulnerable versions of @azure/identity
      Depends on vulnerable versions of @microsoft/teamsfx-core
      Depends on vulnerable versions of tedious
      node_modules/@microsoft/teamsfx-cli
      @microsoft/teamsfx-core  *
      Depends on vulnerable versions of @azure/identity
      Depends on vulnerable versions of @azure/msal-node
      Depends on vulnerable versions of tedious
      node_modules/@microsoft/teamsfx-core
      tedious  >=11.0.9
      Depends on vulnerable versions of @azure/identity
      node_modules/tedious

6 vulnerabilities (5 moderate, 1 high)
```
In order to fix this other Microsoft packages should be fixed e.g. https://github.com/OfficeDev/TeamsFx/blob/dev/packages/cli/package.json

This is too much for me. Should I close the PR?

millerds commented:
We'll have to take care of this by working with the other Microsoft teams.

millerds commented:
Thanks for taking a stab at this @JensMadsen!
